app/src/main/java/app/nexd/android/ui/helper/requestOverview/HelperOverviewViewModel.kt

```diff
@@ -37,24 +33,32 @@ class HelperOverviewViewModel(application: Application) : AndroidViewModel(appli
     }
 
     fun getOtherOpenRequests(): LiveData<List<HelpRequest>> {
-        val observable = reload.flatMap {
-            // get all requests created by other people
-            api.helpRequestsControllerGetAll(
-                null,
-                true,
-                null,
-                true,
-                listOf(
-                    PENDING.value
-                )
-            )
-                .map { list -> list.filter { it.helpListId == null } } // TODO shouldn't be needed
-        }
+        val observable =
+            api.userControllerFindMe()
+                .flatMap { currentUser ->
+                    zipCode = currentUser.zipCode
+                    reloadNearby.flatMap { // get all requests created by other people
+                        api.helpRequestsControllerGetAll(
+                            "me",
+                            true,
+                            listOf(zipCode, ""),
+                            true,
+                            listOf(
+                                PENDING.value
+                            )
+                        )
+                    }
+                }
         return LiveDataReactiveStreams.fromPublisher(observable.toFlowable(BackpressureStrategy.BUFFER))
     }
 
     fun reloadData() {
-        reload.onNext(Unit)
+        reloadAccepted.onNext(Unit)
+        reloadNearby.onNext(Unit)
     }
 
+    fun filterbyZipCode(zipCode: String) {
```

Comment:
i would define this as a BehaviorSubject and flatmap the "other requests" observable on it. this way the data would reload and display automatically as soon as you change the zip code